### ISSUE TITLE
Non-null Int types for "first" argument are not accepted by rule "relay-connection-arguments-spec"

### ISSUE THREAD

**Issue by forabi** [cjoudrey/graphql-schema-linter#148]
The following queries all fail lint checks:

```graphql
  notablePeople(first: Int!, after: ID): NotablePersonConnection!

  users(first: Int!, after: ID, where: UsersQueryWhereInput): UserConnection!
    @requireOneOfRoles(roles: [MODERATOR])

  notablePeopleEvents(
    first: Int!
    after: ID
    where: NotablePersonEventsQueryWhereInput
  ): NotablePersonEventConnection!
```

If I remove the `!`, they all pass. Should not non-null integer types be accepted?

**Comment by forabi** [comment]
Yeah, I totally forgot about backwards navigation. So can we make the rule check that at the integer type is not nullable when only one navigation direction is supported?

**Comment by cjoudrey** [comment]
> So can we make the rule check that the integer type is not nullable when only one navigation direction is supported (or at least accept non-nullable integers)?

Exactly!

If only one pagination direction is supported, I think we should allow non-nullable integers.
